ovos_bus_client/users.py

```diff
@@ -0,0 +1,223 @@
+from ovos_backend_client.api import GeolocationApi
+
+
+def _config2preferences(cfg=None):
+    # TODO - filter relevant keys only, ensure no sensistive or irrelevant info passed along
+    if cfg is None:  # allow empty dict to mean "no preferences"
+        cfg = dict(Configuration.get())
+    return cfg or {}
+
+
+def _config2location(cfg=None):
+    loc = cfg or Configuration.get()["location"]
+    if isinstance(loc, dict):
+        loc = Location.from_dict(cfg)
+    assert isinstance(loc, Location)
+    return loc
+
+
+class Location:
+    geolocation = GeolocationApi()
+
+    def __init__(self, lat, lon):
+        self.latitude = lat
+        self.longitude = lon
+        # everything is automatically derived from lat/lon
+        # this ensures location is not ambiguous
+        self._address = None
+        self._timezone_code = None
+        self._city = None
+        self._state = None
+        self._country = None
+        self._country_code = None
+
+    @property
+    def timezone_code(self):
+        if self._timezone_code is None:
+            self._timezone_code = get_timezone(self.latitude, self.longitude)
+        return self._timezone_code
+
+    @property
+    def timezone(self):
+        if self.timezone_code:
+            return gettz(self.timezone_code)
+        return get_default_tz()
+
+    @property
+    def current_time(self):
+        return datetime.datetime.now(self.timezone)
+
+    @property
+    def address(self):
+        if not self._address:
+            if not self._city or not self._country:
+                self._reverse_geolocate()
+            self._address = self._city
+            if self._state:
+                self._address += ", " + self._state
+            if self._country:
+                self._address += ", " + self._country
+        return self._address
+
+    @property
+    def city(self):
+        if not self._city:
+            self._reverse_geolocate()
+        return self._city
+
+    @property
+    def state(self):
+        if not self._state:
+            self._reverse_geolocate()
+        return self._state
+
+    @property
+    def country(self):
+        if not self._country:
+            self._reverse_geolocate()
+        return self._country
+
+    @property
+    def country_code(self):
+        if not self._country_code:
+            self._reverse_geolocate()
+        return self._country_code
+
+    @classmethod
+    def _reverse_geolocate(cls):
+        data = cls.geolocation.get_reverse_geolocation(lat, lon)
+        loc = cls.from_dict(data)
+        return loc
+
+    @classmethod
+    def from_address(cls, address):
+        data = cls.geolocation.get_geolocation(address) or {}
+        data["address"] = address
+        location = cls.from_dict(data)
+        return location
+
+    @classmethod
+    def from_dict(cls, data):
+        if isinstance(data, str):
+            data = json.loads(data)
+
+        latitude = data["coordinate"]["latitude"]
```

Comment:
Could we take this refactor as a chance to reformat the location dict? I never understood why city/state/country were nested under each other

Reply:
i want to keep compat, this format is used all over...
- personal backend
- selene
- apis in backend-client
- backend manager 
- OCA
- skills
- core
- ??

we will need to live with this format for a long long time, if we deprecate it then its out of scope of this PR

Reply:
but i added a location object, meant to make it easier to work with! once we migrate to the new location class then we can do this more easily